htdocs/fourn/class/fournisseur.commande.class.php

```diff
@@ -2403,14 +2403,21 @@ public function getDispachedLines($status = -1)
 		$ret = array();
 
 		// List of already dispatched lines
-		$sql = "SELECT p.ref, p.label,";
+		$sql = "SELECT p.rowid as pid, p.ref, p.label,";
 		$sql .= " e.rowid as warehouse_id, e.ref as entrepot,";
-		$sql .= " cfd.rowid as dispatchedlineid, cfd.fk_product, cfd.qty, cfd.eatby, cfd.sellby, cfd.batch, cfd.comment, cfd.status, cfd.fk_commandefourndet";
-		$sql .= " FROM ".MAIN_DB_PREFIX."product as p,";
-		$sql .= " ".MAIN_DB_PREFIX."commande_fournisseur_dispatch as cfd";
+		$sql .= " cfd.rowid as dispatchlineid, cfd.fk_product, cfd.qty, cfd.eatby, cfd.sellby, cfd.batch, cfd.comment, cfd.status, cfd.datec";
+		$sql .= " ,cd.rowid, cd.subprice";
+		if ($conf->reception->enabled) {
+			$sql .= " ,cfd.fk_reception, r.date_delivery";
+		}
+		$sql .= " FROM ".MAIN_DB_PREFIX."commande_fournisseur_dispatch as cfd";
+		$sql .= " LEFT JOIN ".MAIN_DB_PREFIX."product as p ON cfd.fk_product = p.rowid";
```

Comment:
Using a LEFT JOIN instead of INNER JOIN increase the cost of the execution. Why not INNER JOIN ?

Reply:
My bad, the left join was part of a specific development, but INNER JOIN is better right now.